peekingduck/pipeline/nodes/draw/instance_mask.py

```diff
@@ -0,0 +1,389 @@
+        self.check_valid_choice(
+            "effect",
+            {"standard", "contrast_brightness", "gamma_correction", "blur", "mosaic"},
+        )
+        self.check_valid_choice("effect_area", {"masked", "unmasked"})
+        _check_type("alpha", float)
+        self.check_bounds("alpha", "[0, +inf]")
+        _check_type("beta", int)
+        self.check_bounds("beta", "[-255, 255]")
+        _check_type("gamma", float)
+        self.check_bounds("gamma", "[0, +inf]")
+        _check_type("blur_kernel_size", int)
+        self.check_bounds("blur_kernel_size", "[1, +inf]")
+        _check_type("mosaic_level", int)
+        self.check_bounds("mosaic_level", "[1, +inf]")
+        _check_type("show_contours", bool)
+        _check_type("contour_thickness", int)
+        self.check_bounds("contour_thickness", "[1, +inf]")
+
+    def run(self, inputs: Dict[str, Any]) -> Dict[str, Any]:
+        """Reads an input image and returns the image (1) with the instance
+        segmentation masks drawn over the detected object instances, or
+        (2) with a visual effect mask applied to it. The visual effect can be
+        applied to the masked areas of the image or to the unmasked areas.
+
+        Args:
+            inputs (dict): Dictionary of inputs with keys "img", "masks,
+            "bbox_labels", "bbox_scores".
+
+        Returns:
+            outputs (dict): Output in dictionary format with key "img".
+        """
+        if self.config["effect"] == "standard":
+            output_img = self._draw_standard_masks(
+                inputs["img"],
+                inputs["masks"],
+                inputs["bbox_labels"],
+                inputs["bbox_scores"],
+            )
+        else:
+            output_img = self._mask_apply_effect(
+                inputs["img"],
+                inputs["masks"],
+            )
+        return {"img": output_img}
+
+    def _draw_standard_masks(  # pylint: disable-msg=too-many-locals
+        self,
+        image: np.ndarray,
+        masks: np.ndarray,
+        bbox_labels: np.ndarray,
+        bbox_scores: np.ndarray,
+    ) -> np.ndarray:
+        """Draws instance segmentation masks over detected objects.
+
+        Args:
+            image (numpy.ndarray): Input image.
+            masks (numpy.ndarray): NumPy array of binary (0/1) masks, one mask
+                for each detected object, in the same order as bbox_labels.
+            bbox_labels (numpy.ndarray): NumPy array of strings representing
+                the labels of detected objects. The order corresponds to
+                ``masks``.
+            bbox_scores (numpy.ndarray): Mask confidence scores. The order
+                corresponds to ``masks``.
+
+        Returns:
+            numpy.ndarray: Input image with instance segmentation masks
+                applied to it.
+        """
+        full_sized_canvas = np.zeros(image.shape, image.dtype)
+        ret_image = image.copy()
```

Comment:
Is `copy()` necessary? It seems to work without it. Same for the other contour drawing methods.

Reply:
Amended.